crates/rethnet_evm_napi/src/block.rs

```diff
@@ -308,7 +308,10 @@ impl Block {
             .iter()
             .map(|transaction| match transaction {
                 rethnet_eth::transaction::SignedTransaction::Legacy(transaction) => {
-                    LegacySignedTransaction::new(&env, transaction).map(Either3::A)
+                    LegacySignedTransaction::from_legacy(&env, transaction).map(Either3::A)
+                }
+                rethnet_eth::transaction::SignedTransaction::EIP155(transaction) => {
+                    LegacySignedTransaction::from_eip155(&env, transaction).map(Either3::A)
```

Comment:
I found this a bit confusing. Are we returning a `LegacySignedTransaction` here on purpose?

Reply:
Yes, javascript doesn't have strongly typed enums. Instead Typescript determines which variant something is based on the presence of variant fields. Given that both the EIP-155 and Legacy versions have the same fields, but different interpretations, we cannot express this in Typescript/Javascript.

I originally did, but it introduced a bug where both Legacy and EIP-155 signed transactions were interpreted as the former (or rather, the first in the list of enum variants that matches all fields)